src/rules/__tests__/no-duplicate-hooks.test.js

```diff
@@ -0,0 +1,311 @@
+'use strict';
+
+const { RuleTester } = require('eslint');
+const rule = require('../no-duplicate-hooks');
+
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 6,
+  },
+});
+
+ruleTester.run('basic describe block', rule, {
+  valid: [
```

Comment:
I think an additional test checking that different kinds of hooks in one describe block are allowed would be good